R/tfpscanner.R

```diff
@@ -62,698 +70,796 @@ tfpscan <- function(tre,
+      logistic_growth_rate = best_gr,
+      logistic_growth_rate_p = lgs$lgrp,
+      simple_logistic_growth_rate = lgs$lgr,
+      gam_logistic_growth_rate = lgs$gam_r,
+      simple_logistic_model_support = lgs$relative_model_support["Logistic"],
+      clock_outlier = .clock_outlier_stat(u, a),
+      lineage = paste(names(sort(table(ulins), decreasing = TRUE)), collapse = "|"),
+      lineage_summary = lineage_summary,
+      cocirc_lineage_summary = cocirc_summary,
+      region_summary = reg_summary,
+      external_cluster = !(u %in% node_ancestors),
+      tips = paste(tu, collapse = "|"),
+      defining_mutations = paste(cmut$defining, collapse = "|"),
+      all_mutations = paste(cmut$all, collapse = "|"),
+      stringsAsFactors = FALSE
+    )
+
+    for (i in seq_along(test_cluster_odds)) {
+      vn <- test_cluster_odds[i]
+      val <- test_cluster_odds_value[i]
+      clodds <- .var_proportionality_stat(u, ta, var = vn, value = val)
+      X <- cbind(X, t(clodds))
+    }
+    rownames(X) <- as.character(u)
+    if (u %in% report_nodes) { # print progress
+      i <- which(nodes == u)
+      message(paste("Progress", round(100 * i / length(nodes)), "%"))
+    }
+    if (detailed_output) {
+      cldir <- glue::glue("{output_dir}/{as.character(u)}")
+      dir.create(cldir, showWarnings = FALSE)
+
+      # summary stat data
+      utils::write.csv(data.frame(statistic = t(X[1, c(
+        "logistic_growth_rate",
+        "simple_logistic_growth_rate",
+        "logistic_growth_rate_p",
+        "gam_logistic_growth_rate",
+        "simple_logistic_model_support",
+        "clock_outlier"
+      )])),
+      file = glue::glue("{cldir}/summary.csv")
+      )
+      # freq plot
+      if (!is.null(lgs$plot)) {
+        suppressMessages(ggplot2::ggsave(lgs$plot,
+          filename = glue::glue("{cldir}/frequency.pdf")
+        ))
+        suppressMessages(ggplot2::ggsave(lgs$plot,
+          filename = glue::glue("{cldir}/frequency.png"),
+          bg = "white"
+        ))
+      }
+
+      # tree plot
+      if (length(tu) < 1e3) {
+        gtr <- .cluster_tree(tu)
+        suppressMessages(
+          ggplot2::ggsave(gtr,
+            filename = glue::glue("{cldir}/clustertree.pdf"),
+            height = max(6, floor(length(tu) / 5)),
+            width = min(44, max(24, sqrt(length(tu)))),
+            limitsize = FALSE
+          )
+        )
+      }
+
+      # clock figure TODO
+      # tip table
+      utils::write.csv(amd[amd$sequence_name %in% tu, ],
+        file = glue::glue("{cldir}/sequences.csv")
+      )
+      # reg summary
+      utils::write.csv(reg_summary,
+        file = glue::glue("{cldir}/regional_composition.csv")
+      )
+      # lineage summary
+      utils::write.csv(lineage_summary,
+        file = glue::glue("{cldir}/lineage_composition.csv")
+      )
+      # cocirc lineage summary
+      utils::write.csv(cocirc_summary,
+        file = glue::glue("{cldir}/cocirculating_lineages.csv")
+      )
```

Comment:
This whole block is now wrapped in `if (detailed_output) {}`, this was introduced in the mrc-ide repo. The code block hasn't really changed, it's just been right-shifted by two spaces. Again, compare against a08bbd6 to see changes against a styled version of the mrc-ide branch.

Reply:
I introduced a change by mistake in the ggplot2::ggsave(..., width = min(44, ...)) line - that has been reverted to width = min(64, ...) as in the mrc-ide repo.